falcon/falcon.py

```diff
@@ -246,231 +295,266 @@ def main(args: Union[str, List[str]] = None) -> int:
     return 0
 
 
-def _prepare_spectra() -> Dict[int, Tuple[int, List[str]]]:
+def _prepare_spectra(
+    process_spectrum: Callable, vectorize: Callable
+) -> Set[int]:
     """
     Read the spectra from the input peak files and partition to intermediate
     files split and sorted by precursor m/z.
 
+    Parameters
+    ----------
+    process_spectrum : Callable
+        The function to process the spectra.
+
     Returns
     -------
-    Dict[int, Tuple[int, List[str]]]
-        A dictionary with per precursor charge the total number of spectra and
-        a list of file names of the spectrum buckets.
+    Set[int]
+        The precursor charges of the spectra.
     """
-    input_filenames = [fn for pattern in config.input_filenames
-                       for fn in glob.glob(pattern)]
-    logger.info('Read spectra from %d peak file(s)', len(input_filenames))
+    input_filenames = [
+        fn for pattern in config.input_filenames for fn in glob.glob(pattern)
+    ]
+    logger.info("Read spectra from %d peak file(s)", len(input_filenames))
     # Use multiple worker processes to read the peak files.
     max_file_workers = min(len(input_filenames), multiprocessing.cpu_count())
-    file_queue = queue.Queue()
     # Restrict the number of spectra simultaneously in memory to avoid
     # excessive memory requirements.
     max_spectra_in_memory = 1_000_000
     spectra_queue = queue.Queue(maxsize=max_spectra_in_memory)
```

Comment:
The queue was slowing down the reading process significantly. The writing part is pretty fast, but I can check if removing this queue will speed it up even more.

Reply:
Writing the spectra to a dict instead of a queue does not improve the runtime
![runtime_queue_vs_dict](https://github.com/user-attachments/assets/f48a158d-7326-471a-8ca0-963b19a6b3c3)